### PR TITLE
Update the documentation for contributors

### PR DESCRIPTION
**Pull request by psss**
Move the develop section to the contribute page.
Mention that tests and documentation are needed.
Add a new section describing the merge process.
Move the Makefile commands section a bit lower.
Plus a couple of minor wording adjustments.

### CODE REVIEW

File: docs/contribute.rst
Comment (FrNecas) [inline docs/contribute.rst:94]: Didn't we discuss some time ago that adding a label meaning "help with tests wanted" would be a good idea? I don't completely remember what was the outcome but it would be worth mentioning if we came to a conclusion

File: docs/contribute.rst
Comment (FrNecas) [inline docs/contribute.rst:164]: I'd perhaps add (maybe not into merging section but somewhere else) a note about the draft feature on github. Something along the lines of "If your pull request is not completely ready but would like to get some early feedback on the concept, create a draft pull request"

Comment (FrNecas) [reply docs/contribute.rst:164]: One more thing - we discussed on one of the previous hacking sessions that the preferred way of fixing a PR after a review is by adding new commits (to avoid having to go through the whole code again) and rebasing/squashing before merge. This could be something worth noting too.

File: docs/contribute.rst
Comment (psss) [reply docs/contribute.rst:94]: Thanks for reminding this. From the notes it seems we agred on a new `tests needed` label. I've just created it and going to add it to the doc.

File: docs/contribute.rst
Comment (psss) [reply docs/contribute.rst:188]: Very good points. Doc updated.

**Review by FrNecas — APPROVED** [review]
LGTM after the changes

**Review by lukaszachy — APPROVED** [review]

**Comment by lukaszachy** [comment]
Well written, nice.

**Comment by psss** [comment]
Thanks for the quick review, squashed and rebased.

**Pull request merged**
